src/blockchain/rpc.rs

```diff
@@ -0,0 +1,664 @@
+                let tx: Transaction = deserialize(&tx_result.hex)?;
+                let mut received = 0u64;
+                let mut sent = 0u64;
+                for output in tx.output.iter() {
+                    if let Ok(Some((kind, index))) =
+                        db.get_path_from_script_pubkey(&output.script_pubkey)
+                    {
+                        if index > *indexes.get(&kind).unwrap() {
+                            indexes.insert(kind, index);
+                        }
+                        received += output.value;
+                    }
+                }
+
+                for input in tx.input.iter() {
+                    if let Some(previous_output) = db.get_previous_output(&input.previous_output)? {
+                        sent += previous_output.value;
+                    }
+                }
+
+                let td = TransactionDetails {
+                    transaction: Some(tx),
+                    txid: tx_result.info.txid,
+                    timestamp: tx_result.info.time,
+                    received,
+                    sent,
+                    fees: tx_result.fee.map(|f| f.as_sat().abs() as u64).unwrap_or(0), //TODO
+                    height: tx_result.info.blockheight,
+                };
+                debug!(
+                    "saving tx: {} tx_result.fee:{:?} td.fees:{:?}",
+                    td.txid, tx_result.fee, td.fees
+                );
+                db.set_tx(&td)?;
+            }
+        }
+
+        for known_txid in known_txs.keys() {
+            if !list_txs_ids.contains(known_txid) {
+                debug!("removing tx: {}", known_txid);
+                db.del_tx(known_txid, false)?;
+            }
+        }
+
+        let current_utxos: HashSet<LocalUtxo> = current_utxo
+            .into_iter()
+            .map(|u| LocalUtxo {
+                outpoint: OutPoint::new(u.txid, u.vout),
+                txout: TxOut {
+                    value: u.amount.as_sat(),
+                    script_pubkey: u.script_pub_key,
+                },
+                keychain: KeychainKind::External,
+            })
+            .collect();
+
+        let spent: HashSet<_> = known_utxos.difference(&current_utxos).collect();
+        for s in spent {
+            debug!("removing utxo: {:?}", s);
+            db.del_utxo(&s.outpoint)?;
+        }
+        let received: HashSet<_> = current_utxos.difference(&known_utxos).collect();
+        for s in received {
+            debug!("adding utxo: {:?}", s);
+            db.set_utxo(s)?;
+        }
+
+        for (keykind, index) in indexes {
+            debug!("{:?} max {}", keykind, index);
+            db.set_last_index(keykind, index)?;
+        }
+
+        self.set_node_synced_height(current_height)?;
+        Ok(())
+    }
+
+    fn get_tx(&self, txid: &Txid) -> Result<Option<Transaction>, Error> {
+        if self.capabilities.contains(&Capability::FullHistory) {
```

Comment:
The docs for `get_tx` are kinda generic and just say "fetch a tx given its txid". Since it's not guaranteed that the tx is part of the wallet, I would check for the `GetAnyTx` capability instead.

Also, maybe it's worth just trying with the node instead of avoiding the req entirely: for unconfirmed txs in its mempool the node should be able to reply even if it doesn't have `txindex` enabled.

Reply:
changed in 653c8735cb55063b92517fa8d727d2ca835afd0c